src/components/molecules/Visualizer/Engine/Cesium/index.tsx

```diff
@@ -99,7 +104,39 @@ const Cesium: React.ForwardRefRenderFunction<EngineRef, EngineProps> = (
           {/* remove default double click event */}
           <ScreenSpaceEvent type={ScreenSpaceEventType.LEFT_DOUBLE_CLICK} />
         </ScreenSpaceEventHandler>
-        <Camera onChange={handleCameraMoveEnd} />
+        <ScreenSpaceCameraController
+          maximumZoomDistance={
+            property?.cameraLimiter?.cameraLimitterEnabled
+              ? property.cameraLimiter?.cameraLimitterTargetArea?.height || Number.POSITIVE_INFINITY
+              : Number.POSITIVE_INFINITY
+          }></ScreenSpaceCameraController>
+        <Camera
+          onChange={handleCameraMoveEnd}
+          percentageChanged={property?.cameraLimiter?.cameraLimitterEnabled ? 0.2 : 0.5}
+        />
+
+        {limiterDimensions && property?.cameraLimiter?.cameraLimitterShowHelper && (
+          <Entity>
+            <PolylineGraphics
+              positions={limiterDimensions.cartesianArray}
+              width={1}
+              material={Color.RED}
+              arcType={ArcType.RHUMB}></PolylineGraphics>
+          </Entity>
+        )}
+        {cameraViewOuterBoundaries && property?.cameraLimiter?.cameraLimitterShowHelper && (
+          <Entity>
+            <PolylineGraphics
+              positions={cameraViewOuterBoundaries.cartesianArray}
+              width={1}
+              material={
+                new PolylineDashMaterialProperty({
+                  color: Color.RED,
+                })
```

Comment:
Material will be recreated and changed every render time. To avoid it, define the material outside the component.

```ts
const cameraViewBoundariesMaterial = new PolylineDashMaterialProperty({
  color: Color.RED,
});
```

Reply:
@rot1024  Should I define it in the hook and use `useMemo` to make sure it gets value once? or It doesn't matter?

Reply:
If `cameraViewBoundariesMaterial` won't be changed by props, just a constant variable is enough. If not, useMemo is good.